flocker/node/agents/test/test_ebs.py

```diff
@@ -189,6 +192,23 @@ def test_unexpected_device_discovered(self, device):
             exception,
         )
 
+    @capture_logging(assertHasMessage, NO_NEW_DEVICE_IN_OS)
+    def test_no_new_device_logged(self, logger):
+        """
+        ``NO_NEW_DEVICE_IN_OS`` is logged if a new device does not appear.
+        """
+        self.assertRaises(
+            AttachedUnexpectedDevice,
+            _attach_volume_and_wait_for_device,
+            volume=self.volume,
+            attach_to=self.compute_id,
+            attach_volume=lambda *a, **kw: None,
+            detach_volume=lambda *a, **kw: None,
+            device=b'/dev/sdh',
+            blockdevices=[],
+            time_limit=0,
+        )
+
```

Comment:
This test relies on the code being able to read some relevant information from `/sys/block`.  So it's subject to interference from other activity on the system.  Since it's only a test for the logging behavior, that's a bit unfortunate.  This is basically a symptom of the factoring of the ebs driver, though, which smears low-level system interactions throughout - so it is hard to avoid having such an interaction in any particular code path.

I'm torn between suggesting we fix the factoring so this doesn't have to be the case and letting this pass for now as a consequence of the existing state of the system.

However, it also occurs to me that if the test exercised `_wait_for_device` directly and `_wait_for_device` parameterized `FilePath("/sys/block")` and `FilePath("/dev")` then this interaction could be avoided...  Probably?